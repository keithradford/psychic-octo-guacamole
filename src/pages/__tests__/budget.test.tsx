import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import BudgetPage, { BudgetSectionTitle } from "../budget";

describe("Budget page", () => {
  it("should render correctly", () => {
    const { container } = render(<BudgetPage />);
    expect(container).toMatchSnapshot();
  });
});

describe("Budget section title", () => {
  it("should display the title", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText("Income")).toBeInTheDocument();
  });

  it("should display the current amount", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText(/\$671/)).toBeInTheDocument();
  });

  it("should display the max amount", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText(/\$1000/)).toBeInTheDocument();
  });
});
