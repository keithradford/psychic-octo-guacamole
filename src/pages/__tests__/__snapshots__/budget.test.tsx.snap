// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget page should render correctly 1`] = `
<div>
  <div
    class="flex flex-col w-full space-y-5"
  >
    <div
      class="flex flex-row flex-grow"
    >
      <div
        class="flex flex-col space-y-1 w-60"
      >
        <h1
          class="text-xl"
        >
          Income
        </h1>
        <p
          class="font-light"
        >
          $
          671
           of $
          1000
        </p>
      </div>
      <div
        class="flex flex-col w-full space-y-2"
      >
        <div
          class="flex items-end justify-between"
        >
          <h1
            class="text-lg"
          >
            Paycheque
          </h1>
          <h2
            class="text-sm font-light"
          >
            $
            671
             of $
            1000
          </h2>
        </div>
        <div
          class="w-full h-4 border border-gray-400 rounded-md"
        >
          <div
            class="bg-green-300 h-full rounded-md"
            data-testid="percent-bar"
            style="width: 67.10000000000001%;"
          />
        </div>
      </div>
    </div>
    <div
      class="w-full border"
    />
    <div
      class="flex flex-row flex-grow"
    >
      <div
        class="flex flex-col space-y-1 w-60"
      >
        <h1
          class="text-xl"
        >
          Spending
        </h1>
        <p
          class="font-light"
        >
          $
          95
           of $
          300
        </p>
      </div>
      <div
        class="flex flex-col w-full space-y-5"
      >
        <div
          class="flex flex-col w-full space-y-2"
        >
          <div
            class="flex items-end justify-between"
          >
            <h1
              class="text-lg"
            >
              Entertainment
            </h1>
            <h2
              class="text-sm font-light"
            >
              $
              15
               of $
              100
            </h2>
          </div>
          <div
            class="w-full h-4 border border-gray-400 rounded-md"
          >
            <div
              class="bg-green-300 h-full rounded-md"
              data-testid="percent-bar"
              style="width: 15%;"
            />
          </div>
        </div>
        <div
          class="flex flex-col w-full space-y-2"
        >
          <div
            class="flex items-end justify-between"
          >
            <h1
              class="text-lg"
            >
              Food
            </h1>
            <h2
              class="text-sm font-light"
            >
              $
              30
               of $
              100
            </h2>
          </div>
          <div
            class="w-full h-4 border border-gray-400 rounded-md"
          >
            <div
              class="bg-green-300 h-full rounded-md"
              data-testid="percent-bar"
              style="width: 30%;"
            />
          </div>
        </div>
        <div
          class="flex flex-col w-full space-y-2"
        >
          <div
            class="flex items-end justify-between"
          >
            <h1
              class="text-lg"
            >
              Transport
            </h1>
            <h2
              class="text-sm font-light"
            >
              $
              50
               of $
              100
            </h2>
          </div>
          <div
            class="w-full h-4 border border-gray-400 rounded-md"
          >
            <div
              class="bg-green-300 h-full rounded-md"
              data-testid="percent-bar"
              style="width: 50%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
